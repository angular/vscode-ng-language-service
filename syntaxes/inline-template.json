{
  "scopeName": "inline-template.ng",
  "injectionSelector": "L:meta.decorator.ts -comment",
  "patterns": [
    {
      "include": "#inline-template"
    }
  ],
  "repository": {
    "inline-template": {
      "begin": "(template)\\s*(:)",
      "beginCaptures": {
        "1": {
          "name": "meta.object-literal.key.ts"
        },
        "2": {
          "name": "meta.object-literal.key.ts punctuation.separator.key-value.ts"
        }
      },
      "end": "(?=,|})",
      "patterns": [
        {
          "include": "#ts-paren-expression"
        },
        {
          "include": "#ng-template"
        }
      ]
    },

    "ts-paren-expression": {
      "begin": "\\G\\s*(\\()",
      "beginCaptures": {
        "1": {
          "name": "meta.brace.round.ts"
        }
      },
      "end": "\\)",
      "endCaptures": {
        "0": {
          "name": "meta.brace.round.ts"
        }
      },
      "patterns": [
        {
          "include": "#ts-paren-expression"
        },
        {
          "include": "#ng-template"
        }
      ]
    },

    "ng-template": {
      "begin": "\\G\\s*([`|'|\"])",
      "beginCaptures": {
        "1": {
          "name": "string"
        }
      },
      "end": "\\1",
      "endCaptures": {
        "0": {
          "name": "string"
        }
      },
      "contentName": "text.html",
      "patterns": [
        {
          "include": "text.html.derivative"
        },
        {
          "include": "template.ng"
        }
      ]
    }
  }
}
