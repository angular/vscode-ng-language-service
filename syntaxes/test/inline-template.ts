/* clang-format off */

@Component({
//// Property key/value test
  template: '<div></div>',

//// String delimiter tests
  template: `<div></div>`,
  template: "<div></div>",
  template: '<div></div>',

//// Parenthesization tests
  template: ( (( '<div></div>' )) ),

//// Comments tests
  // template: '<div></div>'
  /*
   * template: '<div></div>'
   */
  /**
   * template: '<div></div>'
   */
})
export class TMComponent{}
