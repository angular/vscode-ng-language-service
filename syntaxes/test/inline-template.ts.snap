>/* clang-format off */
#^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>
>@Component({
#^^^^^^^^^^^^^ inline-template.ng
>//// Property key/value test
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: '<div></div>',
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>
>//// String delimiter tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: `<div></div>`,
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>  template: "<div></div>",
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>  template: '<div></div>',
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng string
#             ^^^^^^^^^^^ inline-template.ng
#                        ^ inline-template.ng string
#                         ^^ inline-template.ng
>
>//// Parenthesization tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  template: ( (( '<div></div>' )) ),
#^^ inline-template.ng
#  ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#          ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#           ^ inline-template.ng
#            ^ inline-template.ng meta.brace.round.ts
#             ^ inline-template.ng
#              ^ inline-template.ng meta.brace.round.ts
#               ^ inline-template.ng meta.brace.round.ts
#                ^ inline-template.ng
#                 ^ inline-template.ng string
#                  ^^^^^^^^^^^ inline-template.ng
#                             ^ inline-template.ng string
#                              ^ inline-template.ng
#                               ^ inline-template.ng meta.brace.round.ts
#                                ^ inline-template.ng meta.brace.round.ts
#                                 ^ inline-template.ng
#                                  ^ inline-template.ng meta.brace.round.ts
#                                   ^^ inline-template.ng
>
>//// Comments tests
#^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>  // template: '<div></div>'
#^^^^^ inline-template.ng
#     ^^^^^^^^ inline-template.ng meta.object-literal.key.ts
#             ^ inline-template.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#              ^ inline-template.ng
#               ^ inline-template.ng string
#                ^^^^^^^^^^^ inline-template.ng
#                           ^ inline-template.ng string
>  /*
#^^^^^ inline-template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>   */
#^^^^^^ inline-template.ng
>  /**
#^^^^^^ inline-template.ng
>   * template: '<div></div>'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>   */
#^^^^^^ inline-template.ng
>})
#^^^ inline-template.ng
>export class TMComponent{}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-template.ng
>